### DISCUSSION TITLE
게시글 및 프로젝트 관련 확인사항 (메인화면)

### DISCUSSION THREAD

**Discussion by codero225** [80000Coding/80000Coding-Web-Client#166] [General]
안녕하세요~
메인화면 피드에 나오는 게시글/프로젝트 조회화면과 검색 관련되서 확실하지 않은 부분이 있어 확인하려고 해당 디스커션을 열었습니다.
기획자, 프론트 분들 자유롭게 의견달아 주시면 감사하겠습니다

----

<img width="324" alt="image" src="https://github.com/80000Coding/80000Coding-Web-Client/assets/137995283/0eef4365-c384-4e9f-8ede-e560d7cb1815">

1. 게시글 필터 중 그룹이 "게시글의 그룹"인지 "작성자의 그룹"인지 확인할 필요가 있을 것 같습니다.
    - 해당 게시글의 그룹일 경우 : CONTENT 테이블에 해당 field가 없기에 수정필요
    - 작성자가 해당 그룹에 속해있는 경우 : 이 조건으로 검색할 이유가 있을 지
----
2. 프로젝트 검색 및 조회 관련 화면이 필요합니다. 게시글과 똑같은 형식으로 간다고 생각하면 될지, 차이점이 있을 지 확인 부탁드립니다.
----
<img width="211" alt="image" src="https://github.com/80000Coding/80000Coding-Web-Client/assets/137995283/ce7a2059-12ec-4b31-8cfb-5ac8b24ad94e">

3. 전체검색의 결과를 client로 보낼 때 유저, 게시글, 프로젝트를 각각 보내줄 지 한 번에 모아서 보내는게 맞을 지 고민해봐야 할 것 같습니다.
 ----
4. 최신 검색 내역을 "브라우저 캐시"에 저장할 지, "DB"에 저장할 지 결정해야 합니다.
    - 브라우저 캐시에서 저장할 시 : 같은 유저가 다른 기기나 캐시를 삭제하면 볼 수 없음
    - DB에 저장할 시 : 테이블을 새로 설계하여 구성할 필요 있음, 다른 기기에서도 같은 유저라면 볼 수 있음

**Comment by rkskekzzz** [comment]
1. 조직 검색입니다 ㅎㅎ 
apple 아카데미 / 42서울 필터링 시 사용합니다.

2 + 3. 저희는 검색을 통합검색으로 구성합니다. 백엔드 측에서는 각각의 검색 api만 만들어주시면 됩니다 프론트에서 조합합니다. 개별 검색은 따로 없습니다!
4. 로컬 스토리지에 저장할 예정입니다. 백엔드 api 설계는 불필요합니다 ㅎㅎ

**Reply by ChanHoHan** [reply]
(+추가 질문)
1. 조직이 게시글에 적용돼야 한다는 말씀같은데, 지금 현재 게시글을 만들때 조직을 넣는 필드가 없는거 같아요. 제가 이해한게 맞을까요?! ㅎㅎ
